config/main.py

```diff
@@ -2064,6 +2064,538 @@ def delete_snmptrap_server(ctx, ver):
+        return False
+    if any(char in excluded_special_symbols for char in snmp_username):
+        click.echo("SNMP user {} length should be not be greater than 32 characters".format(snmp_username))
+        click.echo("SNMP community string should not have any of these special "
+                   "symbols {}".format(excluded_special_symbols))
+        click.echo("FAILED: SNMP user {} should not have any of these special "
+                   "symbols {}".format(snmp_username, excluded_special_symbols))
+        return False
+    return True
+
+
+def snmp_user_secret_check(snmp_secret):
+    excluded_special_symbols = ['@', ":"]
+    if len(snmp_secret) < 8:
+        click.echo("SNMP user password length should be at least 8 characters")
+        click.echo("SNMP user password length should be not be greater than 64")
+        click.echo("SNMP user password should not have any of these special "
+                   "symbols {}".format(excluded_special_symbols))
+        click.echo("FAILED: SNMP user password length should be at least 8 characters")
+        return False
+    if len(snmp_secret) > 64:
+        click.echo("SNMP user password length should be at least 8 characters")
+        click.echo("SNMP user password length should be not be greater than 64")
+        click.echo("SNMP user password should not have any of these special "
+                   "symbols {}".format(excluded_special_symbols))
+        click.echo("FAILED: SNMP user password length should be not be greater than 64")
+        return False
+    if any(char in excluded_special_symbols for char in snmp_secret):
+        click.echo("SNMP user password length should be at least 8 characters")
+        click.echo("SNMP user password length should be not be greater than 64")
+        click.echo("SNMP user password should not have any of these special "
+                   "symbols {}".format(excluded_special_symbols))
+        click.echo("FAILED: SNMP user password should not have any of these special "
+                   "symbols {}".format(excluded_special_symbols))
+        return False
+    return True
+
+
+@community.command('add')
+@click.argument('community', metavar='<snmp_community>', required=True)
+@click.argument('string_type', metavar='<RO|RW>', required=True)
+@clicommon.pass_db
+def add_community(db, community, string_type):
+    """ Add snmp community string"""
+    string_type = string_type.upper()
+    if not is_valid_community_type(string_type):
+        click.echo("SNMP type must be RO or RW and not {}".format(string_type))
+        sys.exit(1)
+    if not snmp_community_secret_check(community):
+        click.echo("Bad community")
+        sys.exit(2)
+    snmp_communities = db.cfgdb.get_table("SNMP_COMMUNITY")
+    if community in snmp_communities:
+        click.echo("SNMP community {} is already configured".format(community))
+        sys.exit(3)
+    else:
```

Comment:
>else: [](http://example.com/codeflow?start=4&length=5)

you don't need this `else:`, and can reduce one indentation  level of below section.

Reply:
removed.